cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/SqlTransactionRunner.java

```diff
@@ -73,9 +75,12 @@ public void run(TxRunnable runnable) throws TransactionException {
               this.scanFetchSize));
       connection.commit();
     } catch (Exception e) {
-      Throwable cause = e.getCause();
-      if (cause instanceof SQLException) {
-        rollback(connection, new SqlTransactionException((SQLException) cause, e));
+      List<Throwable> causes = Throwables.getCausalChain(e);
```

Comment:
Please add a test to ensure we cover the problem

Reply:
ack.